### PR TITLE
EASY-1271: doi column in bag-index

### PR DESCRIPTION
**Pull request by rvanheest**
Fixes EASY-1271

#### When applied it will...
* add a doi column to the bag-index
* adjust the logic accordingly

@DANS-KNAW/easy for review

### CODE REVIEW

**Comment by rvanheest** [comment]
since @janvanmansum made this project open, I close this PR and make a new one: new commits are not picked up by this PR anymore...

File: src/main/scala/nl/knaw/dans/easy/bagindex/components/BagFacadeComponent.scala [hunk 1]
```diff
@@ -76,5 +79,32 @@ trait Bagit4FacadeComponent extends BagFacadeComponent {
         else Try { UUID.fromString(parts(1)) }
       } else Failure(InvalidIsVersionOfException(bagDir, uri.toASCIIString))
     }
+
+    def getDoi(datasetXML: Path): Try[Doi] = Try {
```
Comment (PaulBoon) [inline src/main/scala/nl/knaw/dans/easy/bagindex/components/BagFacadeComponent.scala]: Maybe we need a new library for handling DDM related actions?

Comment (rvanheest) [inline src/main/scala/nl/knaw/dans/easy/bagindex/components/BagFacadeComponent.scala]: completely agree!

File: src/test/resources/database/bag-index.sql [hunk 2]
```diff
@@ -1,8 +1,9 @@
 create table bag_info (
     bagId text not null unique,
     base text not null,
-    created text not null);
+    created text not null,
+    doi text not null);
```
Comment (PaulBoon) [inline src/test/resources/database/bag-index.sql]: the dog is not unique in this schema and you even add the same one twice, is there a valid case for this which has to do with versions or am I completely lost here?

**Comment by rvanheest** [comment]
@PaulBoon note that this PR is CLOSED!
Please move your comments to #11

File: src/test/scala/nl/knaw/dans/easy/bagindex/components/AddBagToIndexSpec.scala [hunk 3]
```diff
@@ -25,13 +25,14 @@ class AddBagToIndexSpec extends BagIndexDatabaseFixture with AddBagToIndex {
 
   def addBaseTest(): UUID = {
     val bagId = UUID.randomUUID()
+    val doi = "10.5072/dans-x6f-kf66"
```
Comment (PaulBoon) [inline src/test/scala/nl/knaw/dans/easy/bagindex/components/AddBagToIndexSpec.scala]: This doi is used several times, so make it a private val. Also it is one that is in the database (bag-index.sql)

**Pull request closed without merging**